system_tests/bold_challenge_protocol_test.go

```diff
@@ -863,6 +863,142 @@ func create2ndNodeWithConfigForBoldProtocol(
 	return l2client, l2node, assertionChain
 }
 
+// boldMachine wraps a server_arb.MachineInterface.
+type boldMachine struct {
```

Comment:
This isn't test-only code to be clear, it's supposed to be used in real bold chains, but it currently isn't. We should move this code back and then add the createExecutionRun flag (I can help there)